thirdeye/thirdeye-frontend/app/pods/rootcause-entities-cache/service.js

```diff
@@ -8,33 +8,57 @@ export default Ember.Service.extend({
 
   context: null, // {}
 
+  nativeUrns: null, // Set
+
   pending: null, // Set
 
   init() {
     this._super(...arguments);
-    this.setProperties({ entities: {}, context: {}, pending: new Set() });
+    this.setProperties({ entities: {}, context: {}, pending: new Set(), nativeUrns: new Set() });
   },
 
   request(requestContext, urns) {
     console.log('rootcauseEntitiesCache: request()', requestContext, urns);
-    const { context } = this.getProperties('context');
-    if (_.isEqual(context, requestContext)) {
-      console.log('rootcauseEntitiesCache: request: context is up-to-date. ignoring.');
-      return;
-    }
+    const { context, entities, nativeUrns } = this.getProperties('context', 'entities', 'nativeUrns');
 
-    const frameworks = new Set(['relatedEvents', 'relatedDimensions', 'relatedMetrics']);
+    // special case: urn identity
+    const requestNativeUrns = new Set(filterPrefix(urns, 'thirdeye:metric:'));
+    if (!_.isEqual(nativeUrns, requestNativeUrns)) {
+      console.log('rootcauseEntitiesCache: request: refreshing urn identity');
 
-    this.setProperties({ context: _.cloneDeep(requestContext), pending: frameworks });
+      this.setProperties({ nativeUrns: requestNativeUrns });
 
-    frameworks.forEach(framework => {
-      const url = this._makeUrl(framework, requestContext);
-      fetch(url)
-        // .then(checkStatus) // TODO why doesn't this return parsed json here?
-        .then(res => res.json())
-        .then(this._jsonToEntities)
-        .then(incoming => this._complete(requestContext, urns, incoming, framework));
-    });
+      if ([...requestNativeUrns].some(urn => !entities[urn])) {
```

Comment:
let's save this in to a `const` for readability's sake

Reply:
fixed